wwwapp/migrations/0059_migrate_away_from_python_allaccess.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.29 on 2020-03-07 06:29
+from __future__ import unicode_literals
+
+from django.db import migrations
+from allaccess.models import AccountAccess
+from social_django.models import UserSocialAuth
+
+# After the migration is applied in production just comment out code depending on the allaccess library
+def forwards_func(apps, schema_editor):
+    for access in AccountAccess.objects.all():
+        if access.provider.name == 'facebook':
+            UserSocialAuth(provider='facebook', uid=access.identifier, user=access.user, extra_data=access.access_token).save()
+        elif access.provider.name == 'google':
+            UserSocialAuth(provider='google-oauth2', uid=access.user.email, user=access.user, extra_data=access.access_token).save()
```

Comment:
WHAT???? :open_mouth: Why not user id?

And, you take access.user.email here, for one this field is editable by the user,  and you can have different emails for facebook and google...

Reply:
@grzegorz225 @krzys-h the google-oauth2 backend in python-social-login uses the email address as the UID, changing this behaviour would require investigation on where this behavior is internally implemented and either modifying the backend or some step in the auth pipeline.

Do all of our accounts have broken email addresses?
I can try to spend some time in order to change this behaviour but maybe you can propose a better solution?

Reply:
It's not "broken", the ability to change your email to something else than the OAuth provided is a feature...

If python-social-login uses email here, it shouldn't. From google specs:

> email: The user's email address. This value **may not be unique to this user** and **is not suitable for use as a primary key**. Provided only if your scope included the email scope value.

What it should be using is the "sub" field, like django-allaccess did:

> sub (always): An identifier for the user, unique among all Google accounts and never reused. A Google account can have multiple email addresses at different points in time, but the sub value is never changed. **Use sub within your application as the unique-identifier key for the user.** Maximum length of 255 case-sensitive ASCII characters.

Source: https://developers.google.com/identity/protocols/OpenIDConnect

Reply:
@krzys-h I've came across this undocumented flag in python-social-auth which solves this problem:
https://github.com/python-social-auth/social-core/blob/c8d172e3fe5129fe9c7814157684c2d83d7809e9/social_core/backends/google.py#L14